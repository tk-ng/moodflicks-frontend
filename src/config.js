const PosterBaseUrl = "https://image.tmdb.org/t/p/";

export { PosterBaseUrl };
