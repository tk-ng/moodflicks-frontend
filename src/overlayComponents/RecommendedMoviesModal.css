.RecommendedMoviesModal.modal {
	--bs-modal-bg: rgba(152, 0, 0, 0.95);
	background-color: transparent;
	--bs-modal-footer-border-color: black;
	--bs-modal-header-border-color: black;
}

.RecommendedMoviesModal a {
	text-decoration: none;
	text-align: center;
	color: var(--text-color);
}

.offcanvas.offcanvas-end {
	transition: transform 0.3s ease-in-out;
}
