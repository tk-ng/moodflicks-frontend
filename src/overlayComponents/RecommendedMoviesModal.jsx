import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import { MovieCardList } from "../movieComponents/MovieCardList";
import "./RecommendedMoviesModal.css";

export const RecommendedMoviesModal = ({ recMovies, modal, toggle }) => {
	return (
		<div>
			<Modal
				isOpen={modal}
				toggle={toggle}
				fullscreen="sm"
				size="xl"
				centered={true}
				scrollable={true}
				modalClassName="RecommendedMoviesModal"
			>
				<ModalHeader className="text-light">
					Similar Movies to Match This Mood
				</ModalHeader>
				<ModalBody>
					<div className="text-light mb-2">
						These films are frequently grouped with this listâ€™s picks
					</div>
					<MovieCardList movies={recMovies} />
				</ModalBody>
				<ModalFooter>
					<Button color="dark" onClick={toggle}>
						Close
					</Button>
				</ModalFooter>
			</Modal>
		</div>
	);
};
