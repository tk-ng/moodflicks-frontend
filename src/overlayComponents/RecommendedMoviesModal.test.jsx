import { render, screen, waitFor } from "@testing-library/react";
import { RecommendedMoviesModal } from "./RecommendedMoviesModal";
import { MemoryRouter } from "react-router";
import { PosterBaseUrl } from "../config";

describe("Render the RecommendedMoviesModal component", function () {
	const toggle = vi.fn();
	const testMovies = [
		{
			id: "testid1",
			posterPath: "/testposter1.jpg",
			title: "Test Movie 1",
			releaseDate: "2025-02-28",
			overview: "Test Movie 1 Overview.",
			poll: 1,
		},
		{
			id: "testid2",
			posterPath: "/testposter2.jpg",
			title: "G20",
			releaseDate: "Test Movie 2",
			overview: "Test Movie 1 Overview.",
			poll: 1,
		},
	];
	it("mounts without crashing", async function () {
		const { container, asFragment } = render(
			<RecommendedMoviesModal modal={true} toggle={toggle} recMovies={[]} />
		);
		expect(container).toBeInTheDocument();
		expect(asFragment).toMatchSnapshot();
	});

	it("renders with movies", async function () {
		render(
			<MemoryRouter>
				<RecommendedMoviesModal
					modal={true}
					toggle={toggle}
					recMovies={testMovies}
				/>
			</MemoryRouter>
		);
		await waitFor(() =>
			expect(
				screen.getByText("Test Movie 1", { selector: "h5" })
			).toBeInTheDocument()
		);
		screen
			.getAllByAltText("Movie Poster")
			.forEach((el, idx) =>
				expect(el).toHaveAttribute(
					"src",
					PosterBaseUrl + "w185" + `/testposter${idx + 1}.jpg`
				)
			);
	});

	// Without movies the "More Movies Like This" is not displayed. Tested in MoodDetail.test.jsx
});
