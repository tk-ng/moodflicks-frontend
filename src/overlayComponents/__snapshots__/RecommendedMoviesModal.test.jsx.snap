// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render the RecommendedMoviesModal component > mounts without crashing 1`] = `[Function]`;
